tilavarauspalvelu/integrations/email/main.py

```diff
@@ -80,81 +137,64 @@ def send_application_received_email(
 
         email_type = EmailType.APPLICATION_RECEIVED
         context = get_context_for_application_received(language=language)
-        send_emails_in_batches_task.delay(
-            recipients=recipients,
-            subject=context["title"],
-            text_content=render_text(email_type=email_type, context=context),
-            html_content=render_html(email_type=email_type, context=context),
-        )
+        email = EmailData.build(recipients, context, email_type)
+        send_emails_in_batches_task.delay(email_data=email)
 
     @staticmethod
-    def send_application_in_allocation_emails() -> None:
-        """Sends emails to applicants when application round has entered the allocation phase."""
-        applications = Application.objects.should_send_in_allocation_email().order_by("created_date")
-        if not applications:
+    def send_application_section_cancelled(
+        application_section: ApplicationSection,
+        *,
+        language: Lang | None = None,
+    ) -> None:
+        """Sends an email that the whole application section was cancelled by the user"""
+        reservation: Reservation | None = application_section.actions.get_last_reservation()
```

Comment:
Should maybe have a check that the application status is `RESULTS_SENT`?

Reply:
True, just in case somebody does this before the application round results have actually been sent to the applicants. (Even if at the moment the mutation which triggers this is only called from customer UI and should not be available there before that point -- but application section wide series rejection may well be later implemented for staff users.)